src/Symfony/Component/Filesystem/Watcher/INotifyWatcher.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Filesystem\Watcher;
+
+/**
+ * @internal
+ */
+class INotifyWatcher implements WatcherInterface
+{
+    public function watch($path, callable $callback)
```

Comment:
Idea: add a `float $timeout = null` argument to control the max waiting time.
The implementation would use `stream_select()` to pause instead of `inotify_read()`

Reply:
Timeout added to both the filesystem watcher and the inotify watcher, which takes a value in milliseconds